spec/models/user_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# Reference: https://github.com/zquestz/omniauth-google-oauth2#auth-hash
+OMNIAUTH_HASH = {
+  'provider' => 'google_oauth2',
+  'uid' => '100000000000000000000',
+  'info' => {
+    'name' => 'John Smith',
+    'email' => 'john@example.com',
+    'first_name' => 'John',
+    'last_name' => 'Smith',
+    'image' => 'https://lh4.googleusercontent.com/photo.jpg',
+    'urls' => {
+      'google' => 'https://plus.google.com/+JohnSmith'
+    }
+  },
+  'credentials' => {
+    'token' => 'TOKEN',
+    'refresh_token' => 'REFRESH_TOKEN',
+    'expires_at' => 1_496_117_119,
+    'expires' => true
+  },
+  'extra' => {
+    'id_token' => 'ID_TOKEN',
+    'id_info' => {
+      'azp' => 'APP_ID',
+      'aud' => 'APP_ID',
+      'sub' => '100000000000000000000',
+      'email' => 'john@example.com',
+      'email_verified' => true,
+      'at_hash' => 'HK6E_P6Dh8Y93mRNtsDB1Q',
+      'iss' => 'accounts.google.com',
+      'iat' => 1_496_117_119,
+      'exp' => 1_496_117_119
+    },
+    'raw_info' => {
+      'sub' => '100000000000000000000',
+      'name' => 'John Smith',
+      'given_name' => 'John',
+      'family_name' => 'Smith',
+      'profile' => 'https://plus.google.com/+JohnSmith',
+      'picture' => 'https://lh4.googleusercontent.com/photo.jpg?sz=50',
+      'email' => 'john@example.com',
+      'email_verified' => 'true',
+      'locale' => 'en',
+      'hd' => 'company.com'
+    }
+  }
+}.freeze
```

Comment:
It would be best to keep these hashes into https://github.com/nimblehq/ic-rails-huey-toby/tree/develop/spec/fixtures as files

Reply:
@olivierobert I tried to create a file containing the hash, unfortunately, when reading the content file and converting to hash, it requires using `eval` and this will cause of security risk warning.

```ruby
        fixture = file_fixture('omniauth_hash.txt').read
        auth = OmniAuth::AuthHash.new(eval(fixture))
```

We could convert to yaml format but I'd prefer to keep the hash as a reference from the [ominauth auth-hash example.](https://github.com/zquestz/omniauth-google-oauth2#auth-hash). So I think we can leave it as it is for now 🙏

Reply:
P/s: Otherwise, we could use randomzied auth_hash from [Faker](Faker::Omniauth.google )

Reply:
Why does it need to be a YAML file? 🤔  We can load any file type, including JSON:

```
File.read(File.join('spec', 'fixtures', 'public', filename)) 
```

Reply:
@olivierobert We can load the content file, but when we put the content into the `OmniAuth::AuthHash.new` to create a new auth_hash, it will throw an error. Using `eval(file_content)` can resolve the issue unfortunately Rubocop shows a warning about security risks.  

```
NoMethodError:
       undefined method `each_pair' for "{  :provider => \"google_oauth2\",  :uid => \"123456789\",  :info => {    :name => \"John Doe\",    :email => \"john.doe@example.com\",    :first_name => \"John\",    :last_name => \"Doe\",    :image => \"https://lh3.googleusercontent.com/url/photo.jpg\"  },  :credentials => {      :token => \"token\",      :refresh_token => \"another_token\",      :expires_at => 1354920555,      :expires => true  },  :extra => {    :raw_info => {      :sub => \"123456789\",      :email => \"john.doe@example.com\",      :email_verified => true,      :name => \"John Doe\",      :given_name => \"John\",      :family_name => \"Doe\",      :profile => \"https://plus.google.com/123456789\",      :picture => \"https://lh3.googleusercontent.com/url/photo.jpg\",      :gender => \"male\",      :birthday => \"0000-06-25\",      :locale => \"en\",      :hd => \"example.com\"    },    :id_info => {      \"iss\" => \"accounts.google.com\",      \"at_hash\" => \"HK6E_P6Dh8Y93mRNtsDB1Q\",      \"email_verified\" => \"true\",      \"sub\" => \"10769150350006150715113082367\",      \"azp\" => \"APP_ID\",      \"email\" => \"jsmith@example.com\",      \"aud\" => \"APP_ID\",      \"iat\" => 1353601026,      \"exp\" => 1353604926,      \"openid_id\" => \"https://www.google.com/accounts/o8/id?id=ABCdfdswawerSDFDsfdsfdfjdsf\"    }  }}":String
       Did you mean?  each_char
     # ./spec/models/user_spec.rb:12:in `new'
     # ./spec/models/user_spec.rb:12:in `block (4 levels) in <top (required)>'
     # ./spec/support/vcr.rb:31:in `block (2 levels) in <top (required)>'
```

Anw, we will update to use Faker to generate the hash instead of reading from the file 😄